sycl/doc/extensions/proposed/sycl_ext_oneapi_complex_marray.asciidoc

```diff
@@ -0,0 +1,543 @@
+            friend marray operatorOP(const DataT &lhs, const marray &rhs) = delete;
+         
+            // OP is: ==, !=
+            friend marray<bool, NumElements> operatorOP(const DataT &lhs, const marray &rhs) {
+            /* ... */ }
+
+            // OP is: <, >, <=, >=
+            friend marray<bool, NumElements> operatorOP(const DataT &lhs, const marray &rhs)  = delete;
+         
+            friend marray<bool, NumElements> operator!(const marray &v) = delete;
+    }
+
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+The list of deleted operators are: %, %=, ++, --, +, -, &, |, ^, &=, |=, ^=,
+&&, ||, <<, >>, <<=, >>=, <, >, <=, >=, ~, !
+
+The `make_complex_marray` free function is added to construct complex marrays from real and
+imaginary components. Additionally, the free functions `get_real` and
+`get_imag` are added to access the real and imaginary components of the
+`marray` class without modifying the existing `marray` interface. The usage
+of free functions does cause a deviation from the `std::complex` interface.
+However, it does reduce this extensions impact on the `marray` interface.
+
+```C++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+
+  // Make_complex_marray
+
+  template <class T, std::size_t NumElements>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<T, NumElements> &real, const marray<T, NumElements> &imag);
+
+  template <class T, std::size_t NumElements>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<T, NumElements> &real, const T &imag);
+
+  template <class T, std::size_t NumElements>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const T &real, const marray<T, NumElements> &imag);
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<T, NumElements> &real, const marray<T, NumElements> &imag, std::integer_sequence<std::size_t, I...> int_seq);
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<sycl::ext::oneapi::complex<T>, NumElements> &cmplx, std::integer_sequence<std::size_t, I...> int_seq);
+
+  // Get
+
+  // return marray of component
+  template <class T, std::size_t NumElements>
+  marray<T, NumElements> get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input);
+
+  template <class T, std::size_t NumElements>
+  marray<T, NumElements> get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input);
+
+  // return element of component
+  template <class T, std::size_t NumElements>
+  T get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::size_t index);
+
+  template <class T, std::size_t NumElements>
+  T get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::size_t index);
```

Comment:
What are the values in having these two free functions? Could a user not just get the element using `input[index]` and call `imag`/`real` on that?

Reply:
These two functions are for completeness, it would function identically to `arr[index].real()`.

Reply:
Personally I think I would prefer to be without it. I can see the usefulness of the others though. @gmlueck - Thoughts?

Reply:
I think I would leave them out too.  They are more verbose than the existing alternative, so it's not clear what benefit they provide.  Note this applies also to the "set" versions.  For example, consider what they will look like once they get adopted into the `sycl` namespace:

```
float f = m[5].real();
float f = sycl::get_real(m, 5);  // Still more verbose

m[5].real(3.14);
sycl::set_real(m, 5, 3.14);  // Again more verbose
```

Reply:
I don't have any opposition to removing if it feels redundant, so I have removed it.